runtime/lib/backend/null_qubit/NullQubit.hpp

```diff
@@ -40,14 +69,42 @@ namespace Catalyst::Runtime::Devices {
  *   of the device; these are used to implement Quantum Instruction Set (QIS) instructions.
  */
 struct NullQubit final : public Catalyst::Runtime::QuantumDevice {
-    NullQubit(const std::string &kwargs = "{}") {}
-    ~NullQubit() = default; // LCOV_EXCL_LINE
+    NullQubit(const std::string &kwargs = "{}")
+    {
+        auto device_kwargs = Catalyst::Runtime::parse_kwargs(kwargs);
+        if (device_kwargs.find("track_resources") != device_kwargs.end()) {
+            track_resources_ = device_kwargs["track_resources"] == "True";
+        }
+    }
+    ~NullQubit()
+    {
+        if (this->track_resources_) {
+            PrintResourceUsage();
+        }
+    } // LCOV_EXCL_LINE
 
     NullQubit &operator=(const NullQubit &) = delete;
     NullQubit(const NullQubit &) = delete;
     NullQubit(NullQubit &&) = delete;
     NullQubit &operator=(NullQubit &&) = delete;
 
+    /**
+     * @brief Prints resources that would be used to execute this circuit as a JSON
+     */
+    void PrintResourceUsage()
+    {
+        std::cout << RESOURCE_PRINT_DELIMETER << std::endl;
+        std::cout << "{" << std::endl;
+        std::cout << "  \"num_qubits\": " << resource_data_["num_qubits"] << ",\n";
+        std::cout << "  \"num_gates\": " << resource_data_["num_gates"] << ",\n";
+        std::cout << "  \"gate_types\": ";
+        resource_data_.erase("num_gates");
+        resource_data_.erase("num_qubits");
```

Comment:
These lines invalidate `resource_data_` meaning `PrintResourceUsage` cannot be used multiple times. I know this isn't the case, but it could be the cause of future errors when printing resources multiple times for debugging or exploring this feature

Reply:
Fixed in [41f8d28](https://github.com/PennyLaneAI/catalyst/pull/1619/commits/41f8d2898e0983d7e7a6c7353236a99a6b7d8a34)